Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx

```diff
@@ -0,0 +1,181 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   NoiseCalibratorSpec.cxx
+
+#include "CCDB/CcdbApi.h"
+#include "CommonUtils/StringUtils.h"
+#include "DetectorsCalibration/Utils.h"
+#include "MFTCalibration/NoiseCalibratorSpec.h"
+#include "DataFormatsITSMFT/Digit.h"
+#include "DataFormatsITSMFT/CompCluster.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+
+#include "FairLogger.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+
+using namespace o2::framework;
+using namespace o2::utils;
+
+namespace o2
+{
+namespace mft
+{
+
+NoiseCalibratorSpec::NoiseCalibratorSpec(bool useDigits)
+  : mDigits(useDigits)
+{
+}
+
+void NoiseCalibratorSpec::init(InitContext& ic)
+{
+  auto onepix = ic.options().get<bool>("1pix-only");
+  LOG(INFO) << "Fast 1=pixel calibration: " << onepix;
+  auto probT = ic.options().get<float>("prob-threshold");
+  LOG(INFO) << "Setting the probability threshold to " << probT;
+  auto HBperTF = ic.options().get<int>("hb-per-tf");
+  LOG(INFO) << "Nb of HBF per TF used : " << HBperTF;
+
+  mPath = ic.options().get<std::string>("path");
+  mMeta = ic.options().get<std::string>("meta");
+  mStart = ic.options().get<int64_t>("tstart");
+  mEnd = ic.options().get<int64_t>("tend");
+
+  mCalibrator = std::make_unique<CALIBRATOR>(onepix, probT, HBperTF);
```

Comment:
You should set the length of the calibration slot, i.e. in your case infinite (e.g. `std::numeric_limits<TFType>::max`) since you have just 1 slot.

Reply:
Just a comment : by default this PR doesn't use the NoiseSlotCalibrator class but the NoiseCalibrator which doesn't use TimeSlots, as Jouri mentioned during the discussion that in the end ITS doesn't use the former. So I included the appropriate line but commented it out.